json-rpc/docs/client_checklist.md

```diff
@@ -0,0 +1,86 @@
+      - signing, sha3 hashing, address parsing and converting, hex encoding / decoding
+      - LCS utils
+- [ ] JSON-RPC 2.0 Spec:
+  - spec version validation.
+  - batch requests and responses handling.
+- [ ] JSON-RPC client error handling should distinguish the following 3 type errors:
+  - Transport layer error, e.g. HTTP call failure.
+  - JSON-RPC protocol error: e.g. server respond non json data, or can't be parsed into [Libra JSON-RPC SPEC][1] defined data structure, or missing result & error field.
+  - JSON-RPC error: error returned from server.
+- [ ] https
+- [ ] Client connection pool: keep connection alive for less likely getting inconsistent data from connecting to multiple servers.
+- [ ] Handle stale responses:
+  - [ ] client tracks latest server respond block version and timestamp, raise error when received server response contains stale version / timestamp.
+- [ ] Libra User Identifier parsing utils (see [LIP-5][2])
+  - bech32 addresses/subaddresses support
+- [ ] language specific standard release publish: e.g. java maven central repo, python pip
+- [ ] Multi-network: initialize Client with chain id, JSON-RPC server URL
+- [ ] Handle unsigned int64 data type properly
+- [ ] Validate server chain id: client should be initialized with chain id and validate server respond chain id is same.
+- [ ] Validate input parameters, e.g. invalid account address: "kkk". Should return / raise InvalidArgumentError.
+
+# High Level API
+
+- [ ] transfer: wrap peer to peer transfer with metadata script and submit transaction
+  - may have option to wait until transaction executed successfully or failed.
+- [ ] waitForTransactionExecuted(String accountAddress, long sequence, String signedTranscationHash, long timeout):
+  - for given signed transaction sender address, sequence number, expiration time (or 5 sec timeout) to wait and validate execution result is executed, otherwise return/raise an error / flag to tell it is not executed.
+  - when signedTransactionHash validation failed, it should return / raise TransactionSequenceNumberConflictError
+  - when transaction execution vm_status is not "executed", it should return / raise TransactionExecutionFailure
+
+# Read from Blockchain
+
+- [ ] Get metadata
+- [ ] Get currencies
+- [ ] Get events
+- [ ] Get transactions
+- [ ] Get account
+- [ ] Get account transaction
+- [ ] Get account transactions
+- [ ] Get account events
+- [ ] Handle error response
+- [ ] Serialize result JSON to typed data structure
+
+# Submit Transaction
+
+- [ ] Submit [p2p transfer][3] transaction
+- [ ] Submit other [Move Stdlib scripts][4]
+- [ ] Wait for transaction executed:
+  - low level API, consider not expose, only for internal or test usage.
+  - wait for a transaction by get_transaction by account and transaction sequence, no validation of vm_status and signature.
+
+# Testnet support
+
+- [ ] Gen local wallet account keys: private key, public key, auth key
```

Comment:
generate ed25519 private key, derive ed25519 public keys from private key.   generate Single and MultiSig auth-keys